sky/skylet/providers/command_runner.py

```diff
@@ -11,6 +12,25 @@
 from sky.skylet import constants
 
 
+@dataclasses.dataclass
+class DockerLoginConfig:
+    """Config for docker login. Used for pulling from private registries."""
+    username: str
+    password: str
+    repo_uri: str
+    username_key: str = constants.DOCKER_USERNAME_ENV_KEY
+    password_key: str = constants.DOCKER_PASSWORD_ENV_KEY
+    repo_uri_key: str = constants.DOCKER_REPO_URI_ENV_KEY
+
+    @classmethod
+    def from_dict(cls, d: Dict[str, str]) -> 'DockerLoginConfig':
+        return cls(
+            username=d[cls.username_key],
+            password=d[cls.password_key],
+            repo_uri=d[cls.repo_uri_key],
+        )
```

Comment:
How about not saving these keys in the dataclass? Instead we can do:
```
    @classmethod
    def from_env_vars(cls, d: Dict[str, str]) -> 'DockerLoginConfig':
        return cls(
            username=d[constants.DOCKER_USERNAME_ENV_KEY],
            ...
        )
```

In the ray yaml, we can just use the same variable name to save those variables, and in `node_provider.py`, we can just use `DockerLoginConifg(**docker_config['login_config'])`